src/app/MessageDef/ArrayBuilder.cpp

```diff
@@ -0,0 +1,48 @@
+/**
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+/**
+ *    @file
+ *      This file defines Array builder in CHIP interaction model
+ *
+ */
+
+#include "ArrayBuilder.h"
+
+#include <inttypes.h>
+#include <stdarg.h>
+#include <stdio.h>
+
+namespace chip {
+namespace app {
+ArrayBuilder::ArrayBuilder() {}
+
+CHIP_ERROR ArrayBuilder::Init(TLV::TLVWriter * const apWriter, const uint8_t aContextTagToUse)
+{
+    mpWriter = apWriter;
+    mError   = mpWriter->StartContainer(TLV::ContextTag(aContextTagToUse), chip::TLV::kTLVType_Array, mOuterContainerType);
+
+    return mError;
+}
+
+CHIP_ERROR ArrayBuilder::Init(TLV::TLVWriter * const apWriter)
+{
+    mpWriter = apWriter;
```

Comment:
would `return Init(apWriter, TLV::AnonymousTag)` be clearer?

Why do we have a separate Init method instead of having a default value for aContextTagToUse?

Reply:
agree, but they need different parameter type, one is uint8_t, and another is TLV::Tag, which would impact the large PR for command, I would delete Init function with uint8_t in follow-up PR,

Reply:
filed the follow-up ticket, should be easily handled/fixed after IM encode refresh is settle down.)